suite-native/firmware/src/components/UpdateProgressIndicator.tsx

```diff
@@ -0,0 +1,284 @@
+export const UpdateProgressIndicator = ({
+    progress,
+    isError,
+    operation,
+    isStarting,
+}: {
+    progress: number;
+    isError: boolean;
+    operation: FirmwareOperationStatus['operation'];
+    isStarting: boolean;
+}) => {
+    const checkmarkAnimationProgress = useSharedValue(0);
+    const { utils } = useNativeStyles();
+    const progressEnd = useSharedValue(progress / 100);
+    const animatedBackgroundRadius = useSharedValue(0);
+    const backgroundColorFinished = useSharedValue(utils.colors.textPrimaryDefault);
+    const crossSvg = useSVG(require('@suite-common/icons/assets/x.svg'));
+    const trezorLogoSvg = useSVG(require('@suite-common/icons/assets/trezorLogo.svg'));
+    const trezorLogoOpacity = useSharedValue(1);
+    const errorSvgOpacity = useSharedValue(0);
+    const paragraphOpacity = useSharedValue(0);
+    const isSuccess = operation === 'completed';
+    const isInProgress = !isStarting && !isSuccess && !isError;
+    const isDone = isSuccess || isError;
+
+    useEffect(() => {
+        if (isStarting) {
+            animatedBackgroundRadius.value = withTiming(0, { duration: 600 });
+            backgroundColorFinished.value = backgroundColorFinished.value;
+
+            checkmarkAnimationProgress.value = 0;
+            progressEnd.value = withSpring(0);
+
+            trezorLogoOpacity.value = withTiming(1, { duration: 600 });
+            errorSvgOpacity.value = 0;
+            paragraphOpacity.value = 0;
+        }
+        if (isInProgress) {
+            animatedBackgroundRadius.value = withTiming(0, { duration: 600 });
+            backgroundColorFinished.value = backgroundColorFinished.value;
+
+            checkmarkAnimationProgress.value = 0;
+            progressEnd.value = withSpring(progress / 100);
+
+            trezorLogoOpacity.value = 0;
+            errorSvgOpacity.value = 0;
+            paragraphOpacity.value = withTiming(1, { duration: 600 });
+        }
+        if (isSuccess) {
+            animatedBackgroundRadius.value = withSpring(CIRCLE_RADIUS / 2);
+            backgroundColorFinished.value = utils.colors.textPrimaryDefault;
+
+            checkmarkAnimationProgress.value = withDelay(300, withSpring(1));
+            progressEnd.value = 0;
+
+            trezorLogoOpacity.value = 0;
+            errorSvgOpacity.value = 0;
+            paragraphOpacity.value = 0;
+        }
+        if (isError) {
+            animatedBackgroundRadius.value = withSpring(CIRCLE_RADIUS / 2);
+            backgroundColorFinished.value = utils.colors.backgroundAlertRedBold;
+
+            checkmarkAnimationProgress.value = 0;
+            progressEnd.value = 0;
+
+            trezorLogoOpacity.value = 0;
+            errorSvgOpacity.value = withDelay(150, withTiming(1, { duration: 300 }));
+            paragraphOpacity.value = 0;
+        }
+    }, [
+        progress,
+        progressEnd,
+        animatedBackgroundRadius,
+        checkmarkAnimationProgress,
+        isSuccess,
```

Comment:
I think it's not necessary here, useEffect is doing shallow comparisons and since all this values are just primitive types it should not fire effect with every rerender